### PR TITLE
Added educational design principles.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,3 +25,105 @@ Hi, I'm your host, Dima ([Github](https://github.com/dkorolev), [LinkedIn](https
...
+   The next step is the gRPC interface, the de-facto standard of language-agnostic service-to-service communication.
+
+   From there it's uncharted territory. I am down all the way, including raw binary protocols and client libraries for major languages when it comes to setting performance records. It is more important though to get the big picture done right first, and HTTP/JSON is good enough for most intents and purposes, even if they are poor fit for the real large-scale system.
+
+4. **Tested Through**.
+
+   Testing is a big topic, but this list would be incomplete without it. At the very least, the implementation, both of individual components and the whole system, should offer:
+
+   Unit tests: To ensure individual, isolated pieces of logic are correct.
+
```
Comment (Asya-Vorobyova) [inline README.md:78]: Component tests are included into integration ones?

File: README.md [hunk 2]
```diff
@@ -25,3 +25,105 @@ Hi, I'm your host, Dima ([Github](https://github.com/dkorolev), [LinkedIn](https
...
+   Docker is a must have these days, whether we like it or not. Ideally, each component should be designed such that it can be built & run on bare metal, or packed into a Debian module or into a unikernel of its own, or even be coupled with other components into a single combined piece. Still, for the purposes of keeping the design clean, each component should come with a `Dockerfile`, so that a `docker compose` can be natively used to spin up and run & test the entire constellation.
+
+   In particular, if a certain component requires warmup, or has nontrivial dependencies, we implement these in a Docker-native way first, using health checks and additional containers so that the dependency graph is respected by any well-designed test.
+
+   I also want our components to be agnostic with respect to now-common standard building blocks. There should be no "Database" or "Cache" or "Message Queue" or "Leader Elections" components, and there should be no Postgres/MySQL/Mongo, Redis/Memcached, Kafka/RabbitMQ, or Consul/etcd ones. Instead, each component should serve a specific well-articulated logical function.
```
Comment (Asya-Vorobyova) [inline README.md:57]: I didn't get it. Does it mean that we don't want to implement common solutions "from scratch"? Or that we don't want to allot all space of any component for say DB without any additional logic?

File: README.md (hunk 1 above)
Comment (dkorolev) [reply README.md:78]: :+1: 966bd28

File: README.md (hunk 2 above)
Comment (dkorolev) [reply README.md:57]: :+1: 014c279

Comment (dkorolev) [reply README.md:57]: :+1: 014c279

Comment (dkorolev) [reply README.md:57]: :+1: https://github.com/SysDesignMeetup/sdm/commit/014c279

File: README.md [hunk 3]
```diff
@@ -25,3 +25,109 @@ Hi, I'm your host, Dima ([Github](https://github.com/dkorolev), [LinkedIn](https
...
+
+   This should be done such that it is possible to implement the same functionality using multiple underlying building blocks, and effectively compare them to one another, instead of fixating on a particular standard component as the "building block of choice". However tempting it may be, no design we provide should be set on using any particular technology as the backbone. In fact, if a particular technology appears to be a perfect fit, then we likely are either not attacking a generic enough problem, or not thinking of it broadly enough.
+
+3. **Contract-Centric**.
+
+   By contract-centric I mostly mean API-centric, but I keep teaching people to reason in terms of contracts, not APIs. Focusing on the specs of APIs of individual components often is about not seeing the forest behind the trees, and The Contract is a far more encompassing term.
+
+   I would like every module to expose its main interfaces via HTTP/JSON first. This enables demonstrating the core functionality of each individual component from the command line right away, which is a good habit to have, and a great design pattern overall.
```
Comment (chiselko6) [inline README.md:67]: Are you planning to store the changelog? I am referring to some sort of ADR mostly for updated contracts.

Comment (dkorolev) [reply README.md:67]: Of course!

**Comment by AdamEther** [comment]
LGTM! Dima thanks

(I will write a few comments, tho, later)

**Pull request merged**